pennylane/ops/op_math/exp.py

```diff
@@ -0,0 +1,240 @@
+    can be measured as an observable:
+
+    >>> obs = Exp(qml.PauliZ(0), 3)
+    >>> @qml.qnode(qml.device('default.qubit', wires=1))
+    ... def circuit():
+    ...     return qml.expval(obs)
+    >>> circuit()
+    tensor(20.08553692, requires_grad=True)
+
+    """
+
+    coeff = 1
+    """The numerical coefficient of the operator in the exponent."""
+
+    control_wires = Wires([])
+
+    def __init__(self, base=None, coeff=1, do_queue=True, id=None):
+        self.coeff = coeff
+        super().__init__(base, do_queue=do_queue, id=id)
+        self._name = "Exp"
+
+    def __repr__(self):
+        return (
+            f"Exp({self.coeff} {self.base})"
+            if isinstance(self.base, Tensor)
+            else f"Exp({self.coeff} {self.base.name})"
+        )
+
+    @property
+    def data(self):
+        return [[self.coeff], self.base.data]
+
+    @data.setter
+    def data(self, new_data):
+        self.coeff = new_data[0][0]
+        self.base.data = new_data[1]
+
+    @property
+    def num_params(self):
+        return self.base.num_params + 1
+
+    @property
+    def is_hermitian(self):
+        return self.base.is_hermitian and math.imag(self.coeff) == 0
+
+    @property
+    def _queue_category(self):
+        if self.base.is_hermitian and math.real(self.coeff) == 0:
+            return "_ops"
+        return None
+
+    def matrix(self, wire_order=None):
+        def get_mat(op):
+            return qml.matrix(op) if isinstance(op, qml.Hamiltonian) else op.matrix()
+
+        coeff_interface = math.get_interface(self.coeff)
+        if coeff_interface == "autograd":
+            # math.expm is not differentiable with autograd
+            # So we try to do a differentiable construction if possible
+            #
+            # This won't catch situations when the base matrix is autograd,
+            # but at least this provides as much trainablility as possible
+            try:
+                diagonalizing_mat = qml.matrix(self.diagonalizing_gates)()
+                eigvals_mat = math.diag(self.eigvals())
+                mat = diagonalizing_mat.conj().T @ eigvals_mat @ diagonalizing_mat
+            except OperatorPropertyUndefined:
+                warn(
+                    f"The autograd matrix for {self} is not differentiable. "
+                    "Use a different interface if you need backpropagation.",
+                    UserWarning,
+                )
+                base_mat = get_mat(self.base)
+                mat = math.expm(self.coeff * base_mat)
+        else:
+            base_mat = get_mat(self.base)
+            if coeff_interface == "torch":
+                # other wise get `RuntimeError: Can't call numpy() on Tensor that requires grad.`
+                base_mat = math.convert_like(base_mat, self.coeff)
+            mat = math.expm(self.coeff * base_mat)
+
+        return expand_matrix(mat, wires=self.wires, wire_order=wire_order)
```

Comment:
Could we reduce code duplication here?
```suggestion
                mat = diagonalizing_mat.conj().T @ eigvals_mat @ diagonalizing_mat
                return expand_matrix(mat, wires=self.wires, wire_order=wire_order)
            except OperatorPropertyUndefined:
                warn(
                    f"The autograd matrix for {self} is not differentiable. "
                    "Use a different interface if you need backpropagation.",
                    UserWarning,
                )
        base_mat = get_mat(self.base)
        if coeff_interface == "torch":
            # other wise get `RuntimeError: Can't call numpy() on Tensor that requires grad.`
            base_mat = math.convert_like(base_mat, self.coeff)
        mat = math.expm(self.coeff * base_mat)

        return expand_matrix(mat, wires=self.wires, wire_order=wire_order)
```

Reply:
In this case one can also skip the helper method `get_mat`.